pkg/cfn/builder/nodegroup.go

```diff
@@ -185,6 +186,21 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 	return nil
 }
 
+// generateNodeName formulates the name based on the configuration in input
+func (n *NodeGroupResourceSet) generateNodeName() string {
+	name := []string{}
+	if n.spec.InstancePrefix != "" {
+		name = append(name, n.spec.InstancePrefix, "-")
```

Comment:
Omitting the `-` here could add a small amount of flexibility, `acme_` / `acme:`

Reply:
True! However, we may end-up with a "strange" name. For example, let's say that you chose a prefix `foo` and a name `bar`. Without it it will look like `foobar` altogether. Is this something you have in mind? The reason why I put it there was to avoid having the user to write something like

```yaml
instancePrefix: foo-     # the extra - on the end
instanceName: bar
```

I also thought about including a `separator` among the NodeGroup configuration. I wasn't sure if it was actually necessary such a flexibility 😄 . 
Anyway, what do you think we should do in this case:
- add an extra flag called `instanceNameSeparator`
- remove the `-` and leave it without any prefixed separator
- use another separator instead
- ...

Reply:
For sure, thats makes sense, and the prefix would mostly be coupled w/ the generated name which uses `-`s. `instanceNameSeparator` might be overkill :).

Reply:
Shall I leave the code as it is?

Reply:
Yes, I would say so. 👍